sentinel/src/schedule/outbound.js

```diff
@@ -0,0 +1,266 @@
+            },
+            url: urlJoin(config.destination.base_url, authConf.path),
+            json: true
+          };
+
+          return request(authOptions)
+            .then(result => {
+              // No that's not a spelling mistake, this API is sometimes French!
+              if (result.statut !== 200) {
+                logger.error('Non-200 status from Muso auth', result);
+                throw new Error(`Got ${result.statut} when requesting auth`);
+              }
+
+              sendOptions.qs = {
+                token: result.data.username_token
+              };
+            });
+        });
+    }
+
+    // Misconfigured auth type
+    return Promise.reject(new Error(`Invalid auth type '${authConf.type}'. Supported: basic, muso-sih`));
+  };
+
+  return auth().then(() => {
+    if (logger.isDebugEnabled()) {
+      logger.debug('About to send outbound request');
+      logger.debug(JSON.stringify(sendOptions, null, 2));
+    }
+
+    return request(sendOptions)
+      .then(result => {
+        if (logger.isDebugEnabled()) {
+          logger.debug('result from outbound request');
+          logger.debug(JSON.stringify(result, null, 2));
+        }
+      });
+  });
+};
+
+// Collects push configs to attempt out of the task's queue, given a global config
+const getConfigurationsToPush = (config, taskDoc) => {
+  return taskDoc.queue.map(pushName => ([pushName, config[pushName]]));
+};
+
+const singlePush = (taskDoc, medicDoc, config, key) => {
+  const infodoc = configService.getTransitionsLib().infodoc;
+
+  const payload = mapDocumentToPayload(medicDoc, config, key);
+  return send(payload, config)
+    .then(() => {
+      // Worked, remove entry from queue and add link to info doc
+      logger.info(`Pushed ${medicDoc._id} to ${key}`);
+
+      taskDoc.queue.splice(key, 1);
+      if (taskDoc.queue.length === 0) {
+        // Done with this queue completely
+        taskDoc._deleted = true;
+      }
+
+      return db.sentinel.put(taskDoc)
+        .then(({rev}) => {
+          // This works because we are running a single push one at a time and we're passing a
+          // shared reference around (so the next singlePush to run against the same task will have
+          // a reference to this same doc with the updated rev). If we refactor this code to be more
+          // efficient we need to be careful to not cause conflicts against the task document.
+          taskDoc._rev = rev;
+        })
+        .then(() => infodoc.get({id: medicDoc._id, doc: medicDoc}))
+        .then(info => {
+          info.completed_tasks = info.completed_tasks || [];
+          info.completed_tasks.push({
+            type: 'outbound',
+            name: key,
+            timestamp: Date.now()
+          });
+          return db.sentinel.put(info);
+        });
+    })
+    .catch(err => {
+      // Failed
+      logger.error(`Failed to push ${medicDoc._id} to ${key}: ${err.message}`);
+      logger.error(err);
```

Comment:
I think if you want the stack you have to do something like, `logger.error('%o', err);`

Reply:
Or more succinctly, replace both lines with:

```js
      logger.error(`Failed to push ${medicDoc._id} to ${key}: %o`, err);
```

Reply:
Done, though unfortunately it didn't help for 409s due to how those errors get reported (no stack trace)